index.js

```diff
@@ -14,226 +11,226 @@ const getDirName = require('path').dirname;
-                }
-                var tolerance = options.tolerance;
-            }
-            resemble.compare(image1, image2, options, (err, data) => {        
-                if (err) {
-                    reject(err);
-                } else {         
-                    resolve(data);
-                    if (data.misMatchPercentage >= tolerance) {
-                        mkdirp(getDirName(this.config.diffFolder + diffImage), function (err) {
-                            if (err) return cb(err);
-                        });
-                        fs.writeFile(this.config.diffFolder + diffImage + '.png', data.getBuffer(), (err, data) => {
-                            if (err) {
-                                throw new Error(this.err);
-                            }
-                        });
-                    }
-                }
-            });
-        }).catch((error) => {
-            console.log('caught', error.message);
+  /**
+   * Compare Images
+   * 
+   * @param image1
+   * @param image2
+   * @param diffImage
+   * @param options
+   * @returns {Promise<any | never>}
+   */
+  async _compareImages(image1, image2, diffImage, options) {
+    image1 = this.config.baseFolder + image1;
+    image2 = this.config.screenshotFolder + image2;
+
+    return new Promise((resolve, reject) => {
+      if (options.boundingBox !== undefined) {
+        resemble.outputSettings({
+          boundingBox: options.boundingBox
         });
+      }
+
+      this.debug("Tolerance Level Provided " + options.tolerance);
+      const tolerance = options.tolerance;
+
+      resemble.compare(image1, image2, options, (err, data) => {
+        if (err) {
+          reject(err);
+        } else {
+          resolve(data);
+          if (data.misMatchPercentage >= tolerance) {
+            mkdirp(getDirName(this.config.diffFolder + diffImage), function (err) {
+              if (err) return cb(err);
+            });
+            fs.writeFile(this.config.diffFolder + diffImage + '.png', data.getBuffer(), (err, data) => {
+              if (err) {
+                throw new Error(this.err);
+              }
+            });
+          }
+        }
+      });
+    });
+  }
+
+  /**
+   *
+   * @param image1
+   * @param options
+   * @returns {Promise<*>}
+   */
+  async _fetchMisMatchPercentage(image1, options) {
+    const image2 = image1;
+    const diffImage = "Diff_" + image1.split(".")[0];
+    const result = this._compareImages(image1, image2, diffImage, options);
+    const data = await Promise.resolve(result);
+    return data.misMatchPercentage;
+  }
+
+  /**
+   * Take screenshot of individual element.
+   * @param selector selector of the element to be screenshotted 
+   * @param name name of the image
+   * @returns {Promise<void} 
+   */
+  async screenshotElement(selector, name) {
+    const helper = this._getHelper();
+    const configuration = this.config;
+
+    await helper.waitForVisible(selector);
```

Comment:
Add a check here with the helper, 
`if (this.helpers['Puppeteer'])` so that this only tried to screenshot element in case of puppeteer and not for WebdriverIO

Reply:
I've made the changes and added the commit.